pxtcompiler/emitter/driver.ts

```diff
@@ -299,13 +299,42 @@ namespace ts.pxtc {
             includeGreyBlockMessages,
             generateSourceMap: !!opts.ast,
             allowedArgumentTypes: opts.allowedArgumentTypes || ["number", "boolean", "string"],
-            errorOnGreyBlocks: !!opts.errorOnGreyBlocks,
+            errorOnGreyBlocks: !!opts.errorOnGreyBlocks
         };
         const [renameMap, _] = pxtc.decompiler.buildRenameMap(program, file, { declarations: "variables", takenNames: {} })
         const bresp = pxtc.decompiler.decompileToBlocks(blocksInfo, file, decompileOpts, renameMap);
         return bresp;
     }
 
+    // Decompile an array of code snippets (sourceTexts) to XML strings (blocks)
+    export function decompileSnippets(program: Program, opts: CompileOptions, includeGreyBlockMessages = false) {
+        const apis = getApiInfo(program, opts.jres);
```

Comment:
Hmmm... I wonder if we can use lastApiInfo here (and above). Doesn't need to happen in this pr